src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -373,9 +379,13 @@ export class OrganizationIntegrations extends AsyncComponent<
 
   renderBody() {
     const {orgId} = this.props.params;
-    const {displayedList} = this.state;
+    const {displayedList, selectedCategory, list} = this.state;
 
     const title = t('Integrations');
+    const categoryList = list.reduce<string[]>(
+      (acc, curr) => [...new Set([...acc, ...getCategoriesForIntegration(curr)])],
```

Comment:
@NisanthanNanthakumar Not sure why you need a reduce here? Why can't you just use just use lodash's `uniq`?
```
uniq(list.map(getCategoriesForIntegration))
```
Much cleaner

Reply:
@scefali `getCategoriesForIntegration` return `string[]`. So `.map` will return a nested array of categories.

Reply:
@NisanthanNanthakumar right, I think you can use lodash's `flatten` for this:
```
uniq(flatten(list.map(getCategoriesForIntegration)))
```